cvat-core/src/server-proxy.ts

```diff
@@ -1128,42 +1068,22 @@ async function restoreProject(storage: Storage, file: File | string) {
     };
 
     const url = `${backendAPI}/projects/backup`;
-    const projectData = new FormData();
-    let response;
-
-    async function wait() {
-        return new Promise((resolve, reject) => {
-            async function request() {
-                try {
-                    projectData.set('rq_id', response.data.rq_id);
-                    response = await Axios.post(`${backendAPI}/projects/backup`, projectData, {
-                        params,
-                    });
-                    if (response.status === 202) {
-                        setTimeout(request, 3000);
-                    } else {
-                        // to be able to get the task after it was created, pass frozen params
-                        const restoredProject = await getProjects({ id: response.data.id, ...params });
-                        resolve(restoredProject[0]);
-                    }
-                } catch (errorData) {
-                    reject(generateError(errorData));
-                }
-            }
-
-            setTimeout(request);
-        });
-    }
-
     const isCloudStorage = storage.location === StorageLocation.CLOUD_STORAGE;
+    let response;
 
     if (isCloudStorage) {
-        params.filename = file;
-        response = await Axios.post(url,
-            new FormData(), {
-                params,
-            });
-    } else {
+        try {
+            params.filename = file;
+            response = await Axios.post(url,
+                new FormData(), {
+                    params,
+                });
+            return response.data.rq_id;
+        } catch (errorData) {
+            throw generateError(errorData);
+        }
```

Comment:
Since `catch` is the same, I will recommend to use common try in the function body. Please, check other changed functions in this file likewise